feat: 게시글 삭제 기능 추가

dhfkdlsj opened the pull request:
## 개요
- 게시글 삭제(Soft Delete, Hard Delete) 구현
- 게시글 기능 테스트 코드 작성

Resolves: #27(게시글 삭제 기능 추가)

## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

dhfkdlsj committed:
Feat: 게시글 조회 시 삭제되지 않은 글만 조회되는 기능 추가

- 삭제되지않은 게시글만 조회 가능하도록 어노테이션 추가

Resolves: #27

dhfkdlsj committed:
Feat: 게시글 삭제 기능 추가

- 게시글 삭제 기능 추가

Resolves: #27

dhfkdlsj committed:
Feat: 게시글 삭제시 SoftDelete기능 추가

- 게시글 삭제시 SoftDelete 기능 추가
- SoftDelete된 게시글에 대한 복구 기능 추가

Resolves: #27

dhfkdlsj committed:
Fix: 병합시 충돌 해결

- dev 브랜치와의 병합시 충돌 해결

Resolves: #27

dhfkdlsj committed:
Refactor: 주석 제거

- 병합 후 주석 제거

Resolves: #27

dhfkdlsj committed:
Refactor: 주석 추가

- 주석 추가

Resolves: #27

dhfkdlsj committed:
Refactor: 소프트 딜리트 방식 직접 처리 방식으로 변경

- @SQLDelete, @Where 어노테이션 제거하고 삭제 여부 직접 조건 처리
- PostService.getPosts()에서 deleted=false 조건으로 조회
- validateOwner(), validateNotDeleted() 등 검증 로직 Post 엔티티 내부로 이동
    → 서비스 레이어는 도메인 메서드 호출만으로 검증 책임 위임

Resolves: #27

dhfkdlsj committed:
Test: 게시글 기능 테스트

- 게시글 생성, 수정, 조회, 상세조회, 삭제(soft/hard), 복구 테스트 추가
- 삭제된 게시글은 수정 불가 테스트 추가
- 존재하지 않는 게시글 조회 시 예외 발생 테스트 추가
- 작성자가 아닌 사용자의 게시글 수정 시 권한 예외 발생 테스트 추가

Resolves: #27

HaegyeongKim01 commented on src/main/java/com/itcen/censquare/domain/post/service/PostService.java:
```diff
@@ -31,34 +35,69 @@ public Long createPost(PostReqDto request, Member member) {
 
   @Transactional(readOnly = true)
```
getPosts() 메서드에 @Transactional(readOnly = true)를 별도로 지정한 점 좋습니다👏

HaegyeongKim01 reviewed (approved):
LGTM 👍

dhfkdlsj committed:
Refactor: deleted 상태 직접 조회 대신 도메인 메서드로 검증 일원화

- isDeleted() 메서드 제거
- validateDeleted(), validateNotDeleted()를 통해 deleted 상태 검증

Resolves: #27

dhfkdlsj committed:
Refactor: 게시글 상세 조회 메서드에 @Transactional(readOnly = true) 설정

- 게시글 상세 조회 메서드에 @Transactional(readOnly = true) 설정

Resolves: #27

hj-k66 reviewed (commented):
혹시 softDelete된 게시글만 조회하는 API는 나중에 추가되는 걸까요?

dhfkdlsj committed:
Refactor: 사용하지 않는 코드 제거

- 사용하지 않는 코드 제

Resolves: #27
